### PR TITLE
appveyor.yml: Test more than just MSVC2010 + Ninja on x86

### PR DESCRIPTION
**Pull request by nirbheek**
Now we test:
 MSVC 2010 + Ninja (x86)
 MSVC 2015 + Ninja (x86)
 MSVC 2015 + Ninja (x86_64)
 MSVC 2010 + MSBuild (x86)
 MSVC 2015 + MSBuild (x86)
 MSVC 2015 + MSBuild (x86_64)

MSVC 2010 Express only shipped with an x86 toolchain, so we can only test x86 for that.

### CODE REVIEW

**Comment by nirbheek** [comment]
All these failures are failures that we've been missing all this time. I really think that if we advertise support for these platforms and these configurations, we should merge this.

**Comment by nirbheek** [comment]
So with these patches, all failures except for one are now fixed. The last one is not trivial to fix for the general case, so I'll look into that tomorrow.

Once that's fixed, I would urge us to enable this for all builds. Most of these breakages are ~2 months old and it doesn't scale to have one person (me) fix these once every couple of months when I get around to looking at it.

Right now, I don't think CI is a bottleneck for contribution since PRs are merged and reviewed once every 1-2 days anyway. If it starts to become a bottleneck, we can look into enterprise (paid) versions of Travis and Appveyor to get more jobs available to us.

**Comment by jpakkane** [comment]
Great work. I have been away from the Internet for a while so I haven't been able to properly review stuff, but I'll try to get this done ASAP. Thanks.

**Comment by jpakkane** [comment]
The reason linking fails might not be the x86/64 mismatch, but that Python ships optimized and we need to build optimized as well, and I'm not sure if the VS backend does that properly.

**Comment by nirbheek** [comment]
That error would show up as undefined symbols in the CRT. Had to fix that with the 103 manygen test: https://github.com/mesonbuild/meson/pull/995/commits/236f5c2bde7b804fc78d3a45190e05554d748c70

This one was almost certainly because the symbol mangling is different on x86 and x64.

**Comment by jpakkane** [comment]
Diving into the build logs also finds this:

    Traceback (most recent call last):
      File "C:\projects\meson\mesonbuild\mesonmain.py", line 279, in run
        app.generate()
      File "C:\projects\meson\mesonbuild\mesonmain.py", line 169, in generate
        g.generate(intr)
      File "C:\projects\meson\mesonbuild\backend\vs2010backend.py", line 168, in generate
        self.generate_solution(sln_filename, projlist)
      File "C:\projects\meson\mesonbuild\backend\vs2010backend.py", line 233, in generate_solution
        all_deps = self.determine_deps(p)
      File "C:\projects\meson\mesonbuild\backend\vs2010backend.py", line 203, in determine_deps
        all_deps[d.get_id()] = True
    AttributeError: 'GeneratedList' object has no attribute 'get_id'

**Comment by nirbheek** [comment]
Yes, that's the failure I noted that I still have to fix. Haven't had the time to sit down and do that over the weekend.

We're also missing some more tests that will trigger that (and other untested codepaths), and I plan to add those too.

**Comment by nirbheek** [comment]
I've disabled the check for now and opened #1004 about it. I think I'll work on it after the release, and in the meantime we should probably get this in so that we catch problems caused by other PRs early.

**Comment by jpakkane** [comment]
🌩 conflicts.

File: test cases/common/51 pkgconfig-gen/meson.build [hunk 1]
```diff
@@ -19,7 +19,7 @@ pkgg.generate(
 )
 
 pkgconfig = find_program('pkg-config', required: false)
-if pkgconfig.found()
+if pkgconfig.found() and build_machine.system() != 'windows'
```
Comment (jpakkane) [inline test cases/common/51 pkgconfig-gen/meson.build]: This should use the new skipping functionality. Let's merge that first and then this.

Comment (nirbheek) [inline test cases/common/51 pkgconfig-gen/meson.build]: That doesn't skip the whole test though, only the validation of the pkg-config file. I've instead added that to the `113 generatorcustom` test which is indeed wholly skipped with the VS backend.

**Comment by nirbheek** [comment]
Conflicts fixed, and the `113 generatorcustom` test now uses `MESON_SKIP_TEST` in `error()` to signal skipping.

**Pull request merged**